### PR TITLE
C# library v1.2

### PR DESCRIPTION
**Pull request by meliaj**
Mostly autogenerated from swagger-codegen and our swagger.yml, with some code in `override/override.sh` to rename packages/namespaces, copy over some extra files and remind the user to add some changes to the ApiClient.

Events needed some extra work to handle the polymorphism: see `override/CleverEventJsonConverter.cs` and it's usage in `src/Clever/Client/ApiClient.cs`.

I've tested `getStudents` and `getEvents` (see examples on the README) and added a test for events to make sure the override is in place.

cc: @jefff @drhurd who might have some input on C# specifically

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,127 @@
...
+
+namespace Test
+{
+    public static class Program
+    {
+        public static void Main()
+        {
+
+            // Configure OAuth2 access token for authorization: oauth
+            Configuration.Default.AccessToken = "DEMO_TOKEN";
```
Comment (nathanleiby) [inline README.md:24]: Might we instead use something like `GetEnvironmentVariable` here, to remind people not to include/commit secrets in their code? 

https://msdn.microsoft.com/en-us/library/77zkk0b6(v=vs.110).aspx

Same idea for `DEMO_TOKEN` on line 65.

Comment (nathanleiby) [reply README.md:24]: Or at minimum, include a comment around each of these that warns users appropriately around token best-practices

https://dev.clever.com/docs/security#section-security-best-practices

File: README.md [hunk 2]
```diff
@@ -1,2 +1,127 @@
...
+
+            var apiInstance = new DataApi();
+
+            try
+            {
+                StudentsResponse result = apiInstance.GetStudents();
+                foreach (var student in result.Data) {
+                  Console.WriteLine("ID: {0}, Name: {1} {2}", student.Data.Id, student.Data.Name.First, student.Data.Name.Last);
```
Comment (kvigen) [inline README.md:32]: nit: the tabbing / spacing here is weird? or is this standard in c#?

**Review by kvigen — APPROVED** [review]
very nice!

**Pull request merged**